### PR TITLE
feat: Allow specifying node_id

### PR DESCRIPTION
**Pull request by alex-hofsteede**
Same as https://github.com/getsentry/sentry/pull/10399 with an additional change to not generate an id automatically unless we actually have something to save to nodestore. This is so that `bind_nodes` doesn't get confused and try and load something at that id, which was the cause of our 404s in the last round.

I am still not 100% sure where Events with no data payload are being created and then `bind_nodes` called on them. This seems potentially like a logic error somewhere.

### CODE REVIEW

File: src/sentry/db/models/fields/node.py [hunk 1]
```diff
@@ -158,45 +191,60 @@ def on_delete(self, instance, **kwargs):
...
+                if value == {}:
+                    value = None
         else:
-            node_id = None
-            data = value
+            # Either we were passed a null/empty value in the constructor, or
+            # we failed to decode the value from the database so we have no id
+            # to load data from, and no data to save. Set value to empty dict
+            # (not None) so that we don't try and load anything.
+            logger.warn('')
```
Comment (JTCunning) [inline src/sentry/db/models/fields/node.py]: wot

Comment (alex-hofsteede) [reply src/sentry/db/models/fields/node.py]: oops, left some garbage in there.

**Review by mattrobenolt — APPROVED** [review]
wcgw

**Review by JTCunning — APPROVED** [review]
🍪

**Pull request merged**